misc_test.go

```diff
@@ -293,3 +293,48 @@ func TestNaturalLess(t *testing.T) {
 		}
 	}
 }
+
+func TestGetFileExtension(t *testing.T) {
+	dir := t.TempDir()
+	tests := []struct {
+		name              string
+		fileName          string
+		expectedExtension string
+	}{
+		{"normal file", "file.txt", ".txt"},
+		{"file without extension", "file", ""},
+		{"hidden file", ".gitignore", ""},
+		{"hidden file with extension", ".file.txt", ".txt"},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			file, err := os.Create(dir + "/" + test.fileName)
+			if err != nil {
+				t.Fatalf("failed to create file: %v", err)
+			}
```

Comment:
I generally prefer to avoid to avoid having system calls in unit tests. Do you think it would be better to stub the test inputs instead?

Something like this:

```go
type fakeFileInfo struct {
	name  string
	isDir bool
}

func (fileinfo fakeFileInfo) Name() string       { return fileinfo.name }
func (fileinfo fakeFileInfo) Size() int64        { return 0 }
func (fileinfo fakeFileInfo) Mode() os.FileMode  { return os.FileMode(0o000) }
func (fileinfo fakeFileInfo) ModTime() time.Time { return time.Unix(0, 0) }
func (fileinfo fakeFileInfo) IsDir() bool        { return fileinfo.isDir }
func (fileinfo fakeFileInfo) Sys() any           { return nil }

// inside unit test
result := getFileExtension(fakeFileInfo{"file.txt", false})
```

This would also make it possible to test directories with arbitrary names.

Reply:
Good idea! I implemented that now